src/modules/core/components/header/header.tsx

```diff
@@ -1,24 +1,50 @@
 'use client';
-import { type FC } from 'react';
+import { type FC, useEffect, useRef, useState } from 'react';
 import clsx from 'clsx';
 import Link from 'next/link';
+import { createPortal } from 'react-dom';
 // components
 import { Icon } from '@/modules/core/components/icon';
+// types
+import type { ChildrenProp } from '@/modules/core/types/react.types';
 
 import type { HeaderProps } from './header.interface';
 import styles from './header.module.scss';
 
-export const Header: FC<HeaderProps> = ({
+const Header: FC<HeaderProps> & { BottomContent: FC<ChildrenProp> } = ({
   style,
   className,
   rightAdornment,
 }) => {
   return (
     <header className={clsx(styles.root, className)} style={style}>
-      <Link href='/'>
-        <Icon name='stylesync-logo' width={150} />
-      </Link>
-      {rightAdornment}
+      <div className={styles.content}>
+        <Link href='/'>
+          <Icon name='stylesync-logo' width={150} />
+        </Link>
+        {rightAdornment}
+      </div>
+      <div id='header-bottom-node' />
     </header>
   );
 };
+
+const BottomContent: FC<ChildrenProp> = ({ children }) => {
```

Comment:
Could you explain why do you need BottomContent?

Reply:
It's made to attach this progress component to header (see screenshot). 

Why it's done via portal?
1. I don't want this progress component to be part of the header. Header shouldn't know anything about it.
2. This progress component contains some logic that placed in `quizlet-wizard` container. And if we implement progress component as a render prop of header, we need to have state/logic on the top, but even so, header is attached on layout... .
3. It looks comfortable to attach something to header by using `<Header.BottomContent><MyComponentWithSomeState /></Header.BottomContent>` anywhere we what.

<img width="1792" alt="Screenshot 2023-10-07 at 14 05 10" src="https://github.com/StyleSync/stylesync-frontend/assets/35116343/c319e129-5a73-4891-b32a-a4a0f7539b0b">